src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```
@@ -0,0 +1,111 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createStatement =
+                         connection.prepareStatement(
+                             "INSERT INTO manufacturers(name, country) values(?,?);",
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getLong(1);
+                manufacturer.setId(id);
+            }
```

There are 3 problems with this code:

1. This setting ID on existing object is side effect. Each method should have only one responsibility. Method create should not modify manufacturer object, it should only create it in database 😉 
2. Due to the fact that it can be used elsewhere, we should avoid such changes and instead of modifying the object, return a new instance of the object. Such changes may lead to unexpected problems that are difficult to identify in a completely different place in the system, be careful! 😉  Using immutable objects saves us from that. The client could, for example, silence the exception and continue using the manufacturer, which would suddenly have a non-existent ID.
3. We allow the return of manufacturer without ID. What does it mean when we return manufacturer without ID? Does this mean that the creation of manufacturer was performed correctly?

Therefore, I propose:

1. Introducing an immutable manufacturer and returning a new object from this method.
2. Throwing an exception if we do not have the manufacturer ID.